src/flip.sol

```diff
@@ -155,10 +157,12 @@ contract Flipper is LibNote {
         require(lot < bids[id].lot, "Flipper/lot-not-lower");
         require(mul(beg, lot) <= mul(bids[id].lot, ONE), "Flipper/insufficient-decrease");
 
-        vat.move(msg.sender, bids[id].guy, bid);
+        if (msg.sender != bids[id].guy) {
+            vat.move(msg.sender, bids[id].guy, bid);
+            bids[id].guy = msg.sender;
```

Comment:
I think the common case will _not_ be when people are overbidding, so I don't think that lifting the assignment into the conditional is a huge gas save. But I don't see any downside of it, other than it makes the K code more ugly to do so (but I won't be lifting this assignment into the K version I don't think).

Reply:
@ehildenb sorry I just saw the comment after merging it. In `flip` is actually quite common when `dent`ing to overbid, as the first `dent` will probably happen together with the `tend` to get closer to the market price. I understand that anyway this is not a problem for you, but in case it is, let me know and we can revaluate.